engine/src/system.h

```diff
@@ -188,7 +188,7 @@ class MCMemoryFileHandle: public MCSystemFileHandle
 		// If there isn't enough room, extend
 		if (m_pointer + p_length > m_capacity)
```

Comment:
interesting, though, that we're still mixing uint32_t and size_t variables several places in this file.